handlers/handlers.go

```diff
@@ -0,0 +1,369 @@
+
+	if titleOk && title != "" {
+		index.Query(elastic.NewMatchQuery("title", "*"+html.UnescapeString(title)+"*"))
+	}
+	if authorNameOk && authorName != "" {
+		index.Query(elastic.NewMatchQuery("author_name", html.UnescapeString(authorName)))
+	}
+
+	if priceOk {
+		priceF, err := strconv.ParseFloat(price, 64)
+		if err == nil {
+			index.Query(elastic.NewMatchQuery("price", priceF))
+		} else {
+			c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": err.Error()})
+			return
+		}
+
+	}
+
+	if ebookAvailableOk {
+		ebookAvailableB, err := strconv.ParseBool(ebookAvailable)
+		if err == nil {
+			index.Query(elastic.NewMatchQuery("ebook_available", ebookAvailableB))
+		} else {
+			c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": err.Error()})
+			return
+		}
+	}
+
+	if publishDateOk {
+		_, err := time.Parse("2006-01-01", publishDate) // valid time validation
+		if err == nil {
+			index.Query(elastic.NewMatchQuery("publish_date", publishDate))
+		} else {
+			c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": err.Error()})
+			return
+		}
+	}
+
+	result, err := index.Do(c)
+
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
+		return
+	}
+
+	books := make([]models.Book, len(result.Hits.Hits))
+	for i, hit := range result.Hits.Hits {
+		err = json.Unmarshal(hit.Source, &books[i])
+		if err != nil {
+			fmt.Println(err)
+		}
+	}
+
+	c.JSON(http.StatusOK, books)
+}
+
+func (server *Server) Store(c *gin.Context) {
+	query := server.ElasticClient.Search().
+		Index("books").
+		Query(elastic.NewMatchAllQuery())
+
+	titleAggregation := elastic.NewCardinalityAggregation().Field("_id")
+	authorsAggregation := elastic.NewCardinalityAggregation().Field("author_name.keyword")
+	query.Aggregation("number_of_books", titleAggregation)
+	query.Aggregation("number_of_authors", authorsAggregation)
+	query.Size(0)
+
+	results, err := query.Do(c)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": err.Error(),
+		})
+		return
+	}
+
+	var searchResult models.SearchResult
+	res, err := json.Marshal(results.Aggregations)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": err.Error(),
+		})
+		return
+	}
+
+	err = json.Unmarshal(res, &searchResult)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": err.Error(),
+		})
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{
+		"number_of_books":   searchResult.NumberOfBooks.Value,
+		"number_of_authors": searchResult.NumberOfBooks.Value,
+	})
+}
+
+func (server *Server) Activity(c *gin.Context) {
+	username := c.Query("username")
+
+	get, err := server.RedisClient.LRange(username, 0, 2).Result()
+
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": err.Error(),
+		})
+		return
+	}
+
+	var ops [3]models.UserRequest
```

Comment:
Lets init this as an empty slice
```suggestion
	ops := make(models.UserRequest, 0)
```